src/opencl/mod.rs

```diff
@@ -7,14 +7,23 @@ use std::fmt::Write;
 use std::hash::{Hash, Hasher};
 
 pub type BusId = u32;
+
+#[allow(non_camel_case_types)]
 pub type cl_device_id = ocl::ffi::cl_device_id;
```

Comment:
Why not rename to `ClDeviceId`?

Reply:
I did at first, but I think it works out better to use the original OpenCL name. This is especially the case because a consumer needs to transmute values of this type to those of an identical OpenCL type derived differently. Having character-identical names makes it obvious why this is safe.